google-cloud-clients/google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta1/it/ITBigQueryStorageTest.java

```diff
@@ -254,4 +352,174 @@ public void accept(GenericRecord record) {
 
     assertEquals(1_333, avroRowCount);
   }
+
+  @Test
+  public void testReadAtSnapshot() throws InterruptedException, IOException {
+    Field intFieldSchema =
+        Field.newBuilder("col", LegacySQLTypeName.INTEGER)
+            .setMode(Mode.REQUIRED)
+            .setDescription("IntegerDescription")
+            .build();
+    com.google.cloud.bigquery.Schema tableSchema =
+        com.google.cloud.bigquery.Schema.of(intFieldSchema);
+
+    TableId testTableId = TableId.of(/* dataset = */ DATASET, /* table = */ "test_read_snapshot");
+    bigquery.create(TableInfo.of(testTableId, StandardTableDefinition.of(tableSchema)));
+
+    TableReference tableReference =
+        TableReference.newBuilder()
+            .setTableId(testTableId.getTable())
+            .setDatasetId(DATASET)
+            .setProjectId(ServiceOptions.getDefaultProjectId())
+            .build();
+
+    RunQueryJobAndExpectSuccess(
```

Comment:
Consider using the snapshot times from the job's metadata rather than Instant.now().  Using the local system time for snapshots will cause flakiness due to clock drift between the testing instance and the backend, which we've seen in other tests related to snapshot times.

Reply:
@shollyman , feedback applied in commit:
https://github.com/googleapis/google-cloud-java/pull/5242/commits/83a7a95a3e54b85cf83e7d7b10b2b293f81dd5fc